nodeup/pkg/model/kubeapiserver.go

```diff
@@ -70,6 +71,19 @@ func (b *KubeAPIServerBuilder) Build(c *fi.ModelBuilderContext) error {
 		c.AddTask(t)
 	}
 
+	// @check if we are using secure client certificates for kubelet and grab the certificates
+	{
+		if b.UseSecureKubelet() {
```

Comment:
Oh... and mixing this setting is really not going to go well, if the apiserver doesn't have the cert but the nodes are expecting it :-)